src/FEValues/face_integrals.jl

```diff
@@ -1,113 +1,134 @@
+"""
+    face_to_element_transformation(point::Vec{N, T}, cell_T::Type{<:AbstractRefShape}, face::Int) 
+
+Transform point from face's reference (N-1)D coordinates to ND coordinates on the cell's face.
+"""
+face_to_element_transformation
+
+"""
+    weighted_normal(J::AbstractTensor, fv::FaceValues, face::Int)
+    weighted_normal(J::AbstractTensor, cell_T::Type{<:AbstractRefShape}, face::Int)
+
+Compute the vector normal to the face weighted by the area ration between the face and the reference face.
+This is computed by taking cross product of the jacobian compenets that align to the face local axis.
+"""
 function weighted_normal(J::AbstractTensor, fv::FaceValues, face::Int)
     return weighted_normal(J, getrefshape(fv.func_interp), face)
 end
 
+"""
+    create_face_quad_rule(cell_T::Type{RefShape}, w::Vector{T}, p::Vector{Vec{N, T}}
+    create_face_quad_rule(cell_T::Type{RefShape}, quad_faces::Vector{Int}, w_quad::Vector{T}, p_quad::Vector{Vec{N, T}}, tri_faces::Vector{Int}, w_tri::Vector{T}, p_tri::Vector{Vec{N, T}}) 
+
+Creates ["FaceQuadratureRule"](@ref) with the given cell type, weights and points. If the cell has faces of different shapes
+(i.e. quadrilaterals and triangles) then each shape's faces indices, weights and points are passed separately.
+"""
+function create_face_quad_rule(cell_T::Type{RefShape}, w::Vector{T}, p::Vector{Vec{N, T}}) where {N, T, RefShape <: AbstractRefShape}
+    n_points = length(w)
+    face_quad_rule = QuadratureRule{RefShape, T, getdim(AbstractCell{cell_T})}[]
+    for face in 1:nfaces(cell_T)
+        new_points = [face_to_element_transformation(N != 0 ? p[i] : Vec(zero(T)), cell_T, face) for i in 1:n_points] # ξ = 1-t-s, η = s, ζ = 0
```

Comment:
```suggestion
function create_face_quad_rule(::Type{RefShape}, w::Vector{T}, p::Vector{Vec{N, T}}) where {N, T, RefShape <: AbstractRefShape}
    n_points = length(w)
    face_quad_rule = QuadratureRule{RefShape, T, getdim(AbstractCell{RefShape})}[]
    for face in 1:nfaces(RefShape)
        new_points = [face_to_element_transformation(N != 0 ? p[i] : Vec(zero(T)), RefShape, face) for i in 1:n_points] # ξ = 1-t-s, η = s, ζ = 0
```
I think the compiler prefer this pattern.

Also, the comment at the end of the line seems outdated?

Reply:
You mean
```julia
# For cells with mixed faces
```
?
I think it still works because the method underneath is for prisms and pyramids

Reply:
I meant `ξ = 1-t-s, η = s, ζ = 0`

Reply:
Ah, you're right, my bad.